### ISSUE TITLE
Dispatching on native array types fails with optimization switched on

### ISSUE THREAD

**Issue by lizmat** [rakudo/rakudo#4090]
## The Problem

    sub foo(array::intarray \SELF) { dd }
    my int @i = 1,2,3;
    foo(@i);

## Expected Behavior

    sub foo(array::intarray \SELF)

## Actual Behavior

    ===SORRY!=== Error while compiling 11
    Calling foo(Positional[int]) will never work with declared signature (array::intarray \SELF)

Additional notes: running this code with `--optimize=off`, makes it work as expected.  And since the optimizer is responsible for the "cannot ever dispatch" error, this feels like it is incorrectly deciding it will not be able to dispatch, whereas it clearly can.

Additional data-point: changing the call to: `foo($@a)` **also** makes it work, even with the optimization on.  So perhaps it's because it doesn't realize that a native array is already an item?

**Comment by lizmat** [comment]
It looks like in https://github.com/rakudo/rakudo/blob/master/src/Perl6/Optimizer.nqp#L2063 the call to `nqp::p6trialbind` returns `-1`.

**Comment by Altai-man** [comment]
Is it something new?

**Comment by lizmat** [comment]
No, it's probably something very old.  It's just now that I've been able to golf it down  :-)
